chapter01/1.3 - URLify/urlify.js

```diff
@@ -0,0 +1,24 @@
+var urlify = function(str, length) {
+  // have a pointer to check from start to end
+  var strArr = str.split('');
+  var pointer = 0;
+  while (pointer < str.length) {
+    if (strArr[pointer] === ' ') {
+      // *** needs more work here, a little wierd
+      // not handling trailing spaces properly
+      for (var i = str.length - 1; i > pointer + 3; i--) {
+        strArr[i] = str[i - 2];
+      }
+      strArr[pointer] = '%';
+      strArr[pointer+1] = '2';
```

Comment:
The book states "Note: If implementing in Java, please use a character array so that you can perform this operation in place." (pg. 90). I would say the same applies for JavaScript. So use of a char array is appropriate here. In general, when contributing solutions, prefer explicit implementations of relevant algorithms over using built-in libraries even if they are more idiomatic to the language.